smizip/scripts/find_best_ngrams.py

```diff
@@ -191,38 +193,56 @@ def main():
                 singlechars.add(ngram)
             elif len(ngram) > 1:
                 multichars.append(ngram)
-    out.write(f"The initial list of single-char ngrams is:\n  {repr(''.join(sorted(singlechars)))}\n")
-    out.write(f"The initial list of multi-char ngrams is:\n  {multichars}\n")
+    out.write(f"The initial list of single-char n-grams is:\n  {repr(''.join(sorted(singlechars)))}\n")
+    out.write(f"The initial list of multi-char n-grams is:\n  {multichars}\n")
 
     orig_num_ngrams = len(singlechars)
     smiles_iter = open(args.input)
+    
```

Comment:
I had multiple time the issue that the smiles iterator was exhausted before reaching the end else